### PR TITLE
fix: IOU Scan - In dark mode, the damaged PDF - file is barely visible.

### PR DESCRIPTION
**Pull request by Krishna2323**
### Fixed Issues
$ https://github.com/Expensify/App/issues/39356
PROPOSAL: https://github.com/Expensify/App/issues/39356#issuecomment-2029731489

### Tests
1. Open FAB mentu > submit expense > scan
2. Upload corrupt pdf attached below > Select any participant 
3. Verify alert modal is shown with title `Attachment error` 
4. Verify the background has a placeholder with receipt slash component on confirmation page 

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open FAB mentu > submit expense > scan
2. Upload corrupt pdf attached below > Select any participant 
3. Verify alert modal is shown with title `Attachment error` 
4. Verify the background has a placeholder with receipt slash component on confirmation page 

### QA Steps
1. Open FAB mentu > submit expense > scan
2. Upload corrupt pdf attached below > Select any participant 
3. Verify alert modal is shown with title `Attachment error` 
4. Verify the background has a placeholder with receipt slash component on confirmation page 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/85894871/a00a0769-6e56-4a9e-8fac-c7d8abbfd110

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/85894871/a4dd89e3-4ec8-445e-a134-6174efbc771e

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/85894871/6207d5c5-20a6-4983-b358-918caaae763a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/85894871/d6ecb4f8-67e9-4055-9b56-2db83adeaaea

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/85894871/02c98eb5-a5c1-425d-a9a8-f52336f0ae4a

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/85894871/320ace27-7210-4c8c-b4c0-a8f89a37d29e

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
@Krishna2323 When you push the changes for native devices, please ping me.

**Comment by jayeshmangwani** [comment]
@Expensify/design Please confirm the styling of the `Failed to load PDF file.` text?

Screenshot 2024-04-22 at 9 39 17 PM
Screenshot 2024-04-22 at 9 39 32 PM

**Comment by shawnborton** [comment]
I would think instead of showing text like we are, we'd show our default fallback receipt view, something like this that @dannymcclain [had suggested previously](https://github.com/Expensify/App/issues/37435#issuecomment-1995579903):
image

Though I think I would change out the icon to use some kind of broken image icon, something like this but in our brand style:
CleanShot 2024-04-22 at 12 35 56@2x

**Comment by dannymcclain** [comment]
@shawnborton I thought in a different issue we had settled on _not allowing_ someone to upload a corrupted image/PDF? We show the standard modal that says "Error uploading attachment..." This was the mock for that:

image

Am I misremembering something?

**Comment by shawnborton** [comment]
You are definitely right. So yeah, that begs the question, how did we trigger this particular "Failed to load PDF" scenario?

**Comment by jayeshmangwani** [comment]
@shawnborton @dannymcclain To reproduce this issue, I am following this steps:

1. Go to workspace chat > + > Split Expense > Scan > Upload a corrupt pdf

https://github.com/Expensify/App/assets/35371050/1ca7f71b-8b01-4981-9923-150f29d21817

<details>
  <summary>Corrupt pdf</summary>
  
[corrupted.pdf](https://github.com/Expensify/App/files/15067877/corrupted.pdf)

</details>

**Comment by shawnborton** [comment]
Yeah, I feel like a much better solution is to prevent a corrupted PDF from being uploaded in the first place.

**Comment by jayeshmangwani** [comment]
@hayata-suenaga @shawnborton If we are going to prevent a corrupted PDF from being uploaded, then we can close this PR and close/hold the issue in favor of https://github.com/Expensify/App/issues/34032

**Comment by shawnborton** [comment]
That's where my head is at... it seems like we won't ever be able to get to this scenario once we prevent corrupted images/files from being uploaded.

**Comment by trjExpensify** [comment]
>it seems like we won't ever be able to get to this scenario once we prevent corrupted images/files from being uploaded.

I don't think we know the image/file is corrupt purely on the client, so isn't it possible you'll run into this when uploading a corrupt file offline?

**Comment by shawnborton** [comment]
Hmm good question, who can confirm that? 

If that's the case, I like doing something like my suggestion [here](https://github.com/Expensify/App/pull/40607#issuecomment-2070156624).

**Comment by shawnborton** [comment]
Yeah I think any of these would be great! I think I like doing something super generic in case this is a pattern we'd repeat not only for PDF-specific receipts but for all receipts in general that are corrupt. So that being said, I lean towards the first option?

**Comment by dannymcclain** [comment]
Yeah first option makes sense to me—that way we could use it for any file type of receipt that borked.

**Comment by hayata-suenaga** [comment]
> I don't think we know the image/file is corrupt purely on the client, so isn't it possible you'll run into this when uploading a corrupt file offline?

I'm not familiar with how we're handling the PDF upload on the client side, but we try to display a preview of the PDF before the user attempts to send it. If the PDF is corrupted, we know that when we try to display the preview.

@Krishna2323, could you confirm that you seem to be working on the issue to prevent the corrupt PDF from being uploaded?

Screenshot 2024-04-30 at 9 31 25 AM

Regardless, I think it's a good idea to still implement the fallback view if the attachment is corrupted though.

**Comment by hayata-suenaga** [comment]
> Ok, so I guess, we agree on showing fallback for corrupted files, will start implementing the changes today and update.

Yes sorry if that wan't clear 🙇 thank you for pushing the PR forward 👍

**Comment by dannymcclain** [comment]
>I don't think we have Receipt Slash in the app, can you pls share it.

@Krishna2323 No prob!

[Receipt Slash.svg.zip](https://github.com/Expensify/App/files/15237746/Receipt.Slash.svg.zip)

**Comment by shawnborton** [comment]
Hmm I could have sworn we already have a pattern in place for receipt placeholders, like these: 
CleanShot 2024-05-08 at 08 34 07@2x

Can we match that size? From Figma, it looks like they are at 80x80.

**Comment by shawnborton** [comment]
Two questions:

Why can't we detect the corrupted PDF immediately, the same way we do for images?

How did you get a camera view like this where the capture button is off to the right? Looks like we're missing a flash icon? 
CleanShot 2024-05-09 at 07 46 04@2x

**Comment by dannymcclain** [comment]
>Why can't we detect the corrupted PDF immediately, the same way we do for images?

Also curious about this. It feels pretty weird to drop the user back at the contact selector when they acknowledge the warning. They don't need to pick a new contact, they need to upload a different file. So that feels pretty strange to me.

**Comment by shawnborton** [comment]
Hmm yeah, I'm a bit stumped so I would be curious for what our engineering team thinks.

I think the reality is, we haven't had a single customer ever report this corrupted PDF flow. This was a super edge case flow reported by our QA team. So I'm hesitant to spin the wheels too much here, given that I think this is not going to be any kind of mainline flow that we should optimize for. 

If what you are proposing is the best we can do with our current limitations, I think it's probably our best bet.

**Comment by hayata-suenaga** [comment]
> @dannymcclain @shawnborton, we can push the user to file selector page instead of participants selector but we can't detect the pdf immediately. For images we use react-native-image-size package which works on all platforms but we don't have any package to detect corrupt pdf's. We first need to render it with different libraries for native and web and if the render fails we get the error. You can try uploading a protected pdf, you won't get any alert until you reach the confirmation page.

My knowledge of PDF flow is also very limited. But one question. We still display the corrupt PDF on the front end before uploading the PDF to the backend, right? If the file is corrupted, why the front end can display the file?

**Comment by jayeshmangwani** [comment]
@Krishna2323 Remove HOLD from the PR title and resolve the conflicts In local I am getting conflicts with the latest main branch

File: src/components/AttachmentPicker/index.native.tsx
Comment (jayeshmangwani) [inline src/components/AttachmentPicker/index.native.tsx:214]: NAB: If we replace the wording from Image to Attachment for the translate key, should we change the function name to `showAttachmentCorruptionAlert `?

**Comment by jayeshmangwani** [comment]
> resolve the conflicts In local I am getting conflicts with the latest main branch

@Krishna2323 bump for this comment

File: src/components/PDFThumbnail/index.native.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.native.tsx:17]: `hasError` variable name doesn't look good here. Instead we can use something like `failedToLoad` @hayata-suenaga @Krishna2323

File: src/components/PDFThumbnail/index.native.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.native.tsx:22]: Is there any reason you added different conditions when rendering the PDF component and thumbnail?


Why are we checking the hasError in native only and not other platforms?

https://github.com/Expensify/App/blob/31caba2c668877e26dc40a8d91c17d369df24403/src/components/PDFThumbnail/index.native.tsx#L22

https://github.com/Expensify/App/blob/31caba2c668877e26dc40a8d91c17d369df24403/src/components/PDFThumbnail/index.tsx#L57

File: src/components/PDFThumbnail/index.native.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.native.tsx:51]: The lines of code below are common in both files; we can create a common file for this.

```js
<View style={[styles.justifyContentCenter, styles.pdfErrorPlaceholder, styles.alignItemsCenter]}>
    <Icon
        src={Expensicons.ReceiptSlash}
        width={variables.receiptPlaceholderIconWidth}
        height={variables.receiptPlaceholderIconHeight}
        fill={theme.icon}
    />
</View>
```

File: src/components/PDFThumbnail/index.native.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.native.tsx:21]: Why conditional `alignItemsCenter` styling here

File: src/components/PDFThumbnail/index.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.tsx:56]: Why are conditional `alignItemsCenter` and `justifyContentCenter `styling passed here? 

Instead of checking hasError twice for both stylings, please merge them.
```diff
- !hasError && styles.alignItemsCenter, !hasError && styles.justifyContentCenter
+ !hasError && {...styles.alignItemsCenter, ...styles.justifyContentCenter}
```

File: src/components/PDFThumbnail/index.native.tsx
Comment (hayata-suenaga) [reply src/components/PDFThumbnail/index.native.tsx:17]: `failedToLoad` sounds good 😄

File: src/components/PDFThumbnail/index.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.tsx:52]: These conditionals contradict each other. Why should we show a thumbnail when the failedToLoad condition is true??

File: src/components/PDFThumbnail/index.tsx
Comment (jayeshmangwani) [inline src/components/PDFThumbnail/index.tsx:50]: Here, we Added the `styles.h100` parent View Component; why is this change necessary? It was not to the parent View Component before our changes. We don't want to add any visual regression from the PR, so we wanted to make sure it is needed here

**Comment by jayeshmangwani** [comment]
@Krishna2323 Please Update Tests Section something like this:

```js

1. Open App > Press FAB Menu > Submit expense > Select the Scan tab
2. Upload corrupt pdf(attached below) > Select any participant
3. Verify That We can see the Attachment error Modal, and in the background, we can see the [Receipt Slash Image](https://github.com/Expensify/App/pull/40607#issuecomment-2082744344)

```

Below the Tests Section, attach the corrupt pdf file for testing. Also, can you help me understand how I can test these steps? I can test the following:

```js
4. Verify transactions preview also shows receipt slash component
5. Click on transactions preview and verify receipt slash component is also displayed on transaction preview
6. Open transaction details page and verify receipt slash component in place of receipt place holder
```

**Review by jayeshmangwani — APPROVED** [review]
Added a comment for [updating the Tests](https://github.com/Expensify/App/pull/40607#issuecomment-2120348746) section, code changes LGTM

All yours @hayata-suenaga

**Review by hayata-suenaga — APPROVED** [review]
the code change looks good to me. I'm putting HOLD as we have merge freeze put in place. I'll let you know when we lift the merge freeze.

**Comment by hayata-suenaga** [comment]
The merge freeze is being gradually being lifted with the urgent PRs being merged first

I don't think this PR needs to be merged urgently, so let's wait until we have complete merge freeze lift 😄

**Pull request merged**
